pkg/imageset/generate.go

```diff
@@ -23,9 +24,12 @@ type ImageSpec struct {
 }
 
 const (
-	extractImageScript = `#/bin/bash
+	// extractImageScriptTemplate is a minimal shell script we run in the imageset job to determine the images to
+	// use for various OpenShift components involved in the provisioning. A string value will be formatted in
+	// for which installer image we need to use. (regular vs baremetal)
+	extractImageScriptTemplate = `#/bin/bash
 echo "About to run oc adm release info"
-if oc adm release info --image-for="installer" --registry-config "${PULL_SECRET}" "${RELEASE_IMAGE}" > /common/installer-image.txt 2> /common/error.log; then
+if oc adm release info --image-for="%s" --registry-config "${PULL_SECRET}" "${RELEASE_IMAGE}" > /common/installer-image.txt 2> /common/error.log; then
```

Comment:
It would be nice to use [go templates](https://golang.org/pkg/text/template) for this to make it a little easier to maintain, given the length of the template. Although, admittedly, that may be overkill for a template with only a single parameter. With a templates, you could embed in the template the image keys to use and pass the platform as a parameter to the template instead.